[SEC-9320] Add Public Accessibility Guide for CSM (#18833)
DataDog/documentation: content/en/security/cloud_security_management/guide/public-accessibility-logic.md

## Resource Dependency Graph

This graph shows how related resources are used to determine whether other resources are publicly accessible. For example, a Cloudtrail Trail stored in a public S3 bucket is itself publicly accessible. If a resource is publicly accessible because of another resource, we will show this relation in the CSPM Resource relationships graph.

Use this graph as a reference for how resources are correlated to determine public accessibility. Note: not all resources with the Publicly Accessible attribute are shown in this graph:

{{< img src="security/cloud_security_management/guide/public_accessibility_relationships.png" alt="A graph diagram showing the relationships between resources that are used to determine public accessibility" width="75%">}}

## Public Accessibility Logic by Resource

The following section details on a resource-by-resource basis the logic behind public accessibility determination for a resource. 
**Note:** not all resources are able to be classified as publicly accessible or not publicly accessible through only their resource configuration. Such resources are not included here.

### Amazon S3 Bucket

> An [S3 Bucket][1] (`aws_s3_bucket`) is considered is considered publicly accessible if:

Reviewer: ```suggestion
An [S3 Bucket][1] (`aws_s3_bucket`) is considered publicly accessible if:
```

---
title: Background for Public Accessibility
kind: guide
further_reading:
- link: "/security/cspm"
  tag: "Documentation"
  text: "Cloud Security Posture Management"
- link: "/security/default_rules/#cat-cloud-security-management"
  tag: "Documentation"
  text: "Out-of-the-box Detection Rules"
---

Datadog uses a graph processing framework to map relationships between cloud resources to determine whether they are accessible from the internet. This guide outlines the logic used to classify resources as publicly accessible within the graph framework. 

For more information on network reachability, see the [AWS documentation][34] and the [AWS Network Reachability Analyser][35]. Currently, the `Is Publicly Accessible` facet is only available for AWS resources.

## Resource Dependency Graph

> The following diagram shows how related resources are used to determine whether other resources are publicly accessible. For example, a Cloudtrail Trail stored in a public S3 bucket is itself publicly accessible. If a resource is publicly accessible because of another resource, we will show this relation in the CSPM Resource relationships graph.

Reviewer 2: ```suggestion
The following diagram shows how related resources are used to determine whether other resources are publicly accessible. For example, a Cloudtrail Trail stored in a public S3 bucket is itself publicly accessible. If a resource is publicly accessible because of another resource, the relationship is shown in the CSPM Resource relationships graph.
```

|The bucket’s grant list contains a uri value of `http://acs.amazonaws.com/groups/global/AllUsers` or `/AuthenticatedUsers`. |`AllUsers` gives anyone in the world access to the bucket. `AuthenticatedUsers` gives any AWS authenticated user in the world access to the bucket. |

***OR***

* _Bucket Policy-determined access:_

| **Criteria** | **Explanation** |
|--------------|-----------------|
|The bucket’s `public_access_block_configuration` AND the bucket account’s public access block (`aws_s3_account_public_access_block`) both have `ignore_public_acls` set to `false`. |An Access Control List (ACL) defines the AWS accounts and groups that are granted access to this bucket. With `ignore_public_acls` set to `false`, the bucket's configuration permits the use of an ACL that allows public access.  |

> |The bucket’s policy statement allows the `s3:GetObject` permission unconditionally, with resource and principal set to `"*"`. |This defines a public policy on the bucket, meaning that cross-account access is allowed. `"*"` is a wildcard, meaning access is given to any resource and principal. |

Reviewer 2: ```suggestion
|The bucket's policy statement allows the `s3:GetObject` permission unconditionally, with resource and principal set to `"*"`. |This defines a public policy on the bucket, meaning that cross-account access is allowed. `"*"` is a wildcard, meaning access is given to any resource and principal. |
```

|It's in a public [subnet][4].|-|
|It's associated with a [security group][12] that has rules allowing access from a CIDR range of `"0.0.0.0/0"`, or an IPv6 CIDR range of `"::/0"`. |A security group controls inbound traffic to a VPC. With an open CIDR range, all IP addresses are able to gain access. |

***OR***

* _ELB-determined access through autoscaling group:_

| **Criteria** | **Explanation** |
|--------------|-----------------|
|A security group (call it `SG1`) attached to the load balancer is publicly accessible and allows ingress traffic to some port `X`|This opens the load balancer to incoming traffic from the internet on a specific port|
|The load balancer has a listener accepting traffic on port `X`|A [listener][37] is a process that checks for connection requests, using the protocol and port that you configure|
|The load balancer has a target group forwarding traffic to some port `Y`|[Target groups][38] route requests to one or more registered targets, such as EC2 instances, on a protocol and port that you specify|
|An autoscaling group is attached to the load balancer’s target group|-|

> |The EC2 instance is part of the autoscaling group, and has a security group that has at least one rule that allows ingress traffic from port `Y`, either from `0.0.0.0/0`, from the CIDR of the VPC (e.g. `10.0.0.0/8`), or from the security group of the load balancer (`SG1`).|This opens the EC2 instance to traffic coming from the load balancer. The security group must allow traffic from the load balancer, and thus must be open either to all IPs, all IPs in the VPC, or that specific security group.|

Reviewer 2: ```suggestion
|The EC2 instance is part of the autoscaling group, and has a security group that has at least one rule that allows ingress traffic from port `Y`, either from `0.0.0.0/0`, from the CIDR of the VPC (for example, `10.0.0.0/8`), or from the security group of the load balancer (`SG1`).|This opens the EC2 instance to traffic coming from the load balancer. The security group must allow traffic from the load balancer, and thus must be open either to all IPs, all IPs in the VPC, or that specific security group.|
```
